easybuild/tools/options.py

```diff
@@ -1178,6 +1179,16 @@ def process_software_build_specs(options):
                 value = value.split(',')
             build_specs.update({param: value})
 
+    # process --extra-toolchainopts
+    if options.extra_toolchainopts:
+        try_to_generate = True
+
+        toolchainopts = {}
+        for spec in options.extra_toolchainopts:
+            param, value = spec.split('=')
+            toolchainopts.update({param: value})
+        build_specs.update({'toolchainopts': toolchainopts})
+
```

Comment:
In case someone uses --extra-toolchains without arguments, is this caught somewhere?
If not, only set try_to_generate if toolchainopts is actually non-empty, and don't call build_specs.update if it is empty.

Or perhaps error out if it is empty?

Reply:
thanks for looking into this

if it is empty the options parser will error out saying that option it requires an argument

but if it is malformed, that's another issue, for instance if there is no '=' the split above will fail with no more values to unpack...

the correct usage allows something like `--extra-toolchainopts=usempi=True,openmp=True,optarch="Intel:-O2;GCC=-O3"`, maybe this should be made clearer somewhere?